lib/infra/log/log.go

```diff
@@ -0,0 +1,121 @@
+package log
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// Level indicates the severity of the data being logged
+type Level string
+
+var (
+	// LevelDebug are debug or trace information
+	LevelDebug Level = "DEBUG"
+	// LevelInfo are routine information
+	LevelInfo Level = "INFO"
+	// LevelWarning warns about events the might cause problems to the system
+	LevelWarning Level = "WARNING"
+	// LevelError alerts about events that are likely to cause problems
+	LevelError Level = "ERROR"
+	// LevelCritical alerts about severe problems. Most of the time, needs some human intervention ASAP
+	LevelCritical Level = "CRITICAL"
+)
+
+var levelPriority = map[Level]int{
+	LevelCritical: 5,
+	LevelError:    10,
+	LevelWarning:  15,
+	LevelInfo:     20,
+	LevelDebug:    25,
+}
+
+// LogAttribute represents an information to be extracted from the context and included into the log
+type LogAttribute string
+
+// LogAttributeSet is a set of LogAttributes
+type LogAttributeSet map[LogAttribute]bool
+
+// LoggerInput defines the dependencies of a Logger
+type LoggerInput struct {
+	Level      Level
+	Attributes LogAttributeSet
+}
+
+// Logger is the structure responsible for log data
+type Logger struct {
+	level      Level
+	attributes LogAttributeSet
+}
+
+// NewLogger constructs a new Logger instance
+func NewLogger(in LoggerInput) *Logger {
+	if _, ok := levelPriority[in.Level]; !ok {
+		in.Level = LevelInfo
+	}
+
+	return &Logger{level: in.Level, attributes: in.Attributes}
+}
+
+// Debug logs debug data
+func (l Logger) Debug(ctx context.Context, msg string, args ...interface{}) {
+	if levelPriority[l.level] >= 25 {
```

Comment:
Thus, you can do this:
```suggestion
	if l.level >= LevelDebug {
```